engine/api/application.go

```diff
@@ -524,20 +524,22 @@ func updateApplicationHandler(w http.ResponseWriter, r *http.Request, db *sql.DB
 	app, err := application.LoadApplicationByName(db, projectKey, applicationName)
 	if err != nil {
 		log.Warning("updateApplicationHandler> Cannot load application %s: %s\n", applicationName, err)
-		w.WriteHeader(http.StatusNotFound)
+		WriteError(w, r, err)
 		return
 	}
 
 	var appPost sdk.Application
 	// Get body
 	data, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
+		log.Warning("updateApplicationHandler> Cannot read body: %s\n", err)
+		WriteError(w, r, err)
```

Comment:
i suggest to keep WriteHeader, because il the body can't be read, status code should be 400 and not 500